### DISCUSSION TITLE
How to create Endpoints for upfront unknown ports

### DISCUSSION THREAD

**Discussion by pfeigl** [microsoft/aspire#6691] [Q&A] [answered]
We have an application which when started opens a socket server and waits for TCP connections. This service is booted with multiple replicas.
Every replica which starts needs to talk to a defined service and retrieve information on which port the current instance should listen. The service which gives out the ports which should be used is outside of the Aspire stack.

Now in the Aspire Dashboard, I'd like to see the endpoints of the started replicas. (and ideally only after the endpoint is actually reachable, but thats not a must).

Is there any way to feed the information under which port the application is reachable back into the Aspire stack?
I was looking into the eventing model of Aspire, but was not sure whether this fits the usecase and even if, how to update the endpoint for a given resource.

Thanks for any hints / tips,
Philipp

**Comment by davidfowl** [comment, marked as answer]
There's no way to update the list of endpoints for an existing project or container dynamically. Endpoints are determined before the model is  started.

**Comment by pfeigl** [comment]
Too bad, but thanks for the update!
